src/main/java/example/GSAJaccardSimilarityMeasure.java

```diff
@@ -0,0 +1,201 @@
+package example;
+
+import org.apache.flink.api.common.ProgramDescription;
+import org.apache.flink.api.common.functions.FlatMapFunction;
+import org.apache.flink.api.common.functions.MapFunction;
+import org.apache.flink.api.common.functions.ReduceFunction;
+import org.apache.flink.api.java.DataSet;
+import org.apache.flink.api.java.ExecutionEnvironment;
+import org.apache.flink.api.java.tuple.Tuple2;
+import org.apache.flink.graph.Edge;
+import org.apache.flink.graph.EdgeDirection;
+import org.apache.flink.graph.Graph;
+import org.apache.flink.graph.ReduceNeighborsFunction;
+import org.apache.flink.graph.Triplet;
+import org.apache.flink.graph.Vertex;
+import org.apache.flink.types.NullValue;
+import org.apache.flink.util.Collector;
+import util.JaccardSimilarityMeasureData;
+
+import java.util.HashMap;
+import java.util.HashSet;
+
+public class GSAJaccardSimilarityMeasure implements ProgramDescription {
+
+	public static void main(String [] args) throws Exception {
+
+		if (!parseParameters(args)) {
+			return;
+		}
+
+		ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
+
+		DataSet<Edge<Long, NullValue>> edges = getEdgesDataSet(env);
+
+		// initialize the vertex values with empty sets
+		Graph<Long, Tuple2<HashSet<Long>, HashMap<Long, Double>>, NullValue> graph = Graph.fromDataSet(edges,
+				new MapFunction<Long, Tuple2<HashSet<Long>, HashMap<Long, Double>>>() {
+
+					@Override
+					public Tuple2<HashSet<Long>, HashMap<Long, Double>> map(Long id) throws Exception {
+						HashSet<Long> neighbors = new HashSet<Long>();
+						neighbors.add(id);
+
+						return new Tuple2<HashSet<Long>, HashMap<Long, Double>>(neighbors,
```

Comment:
At some point, you asked to have the entire set of computed Jaccard values in the vertex value. That way the skewed vertex will be overwhelmed, which is pretty much what we want to prove.